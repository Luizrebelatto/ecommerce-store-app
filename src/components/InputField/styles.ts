import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
    input: {
      borderWidth: 1,
      borderColor: "#ccc",
      borderRadius: 8,
      padding: 10,
      fontSize: 16,
      marginBottom: 10,
    },
  });
  