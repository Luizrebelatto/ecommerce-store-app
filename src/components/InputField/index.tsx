import React from "react";
import { TextInput, Text } from "react-native";
import { styles } from "./styles";

interface IInputField {
    value: string;
    placeholder?: string;
    onChangeText?: (text: string) => void;
    label?: string;
}

export default function InputField(data: IInputField){
    return(
        <>
            <Text style={styles.label}>{data.label}</Text>
            <TextInput
                value={data.value}
                onChangeText={data.onChangeText}
                style={styles.textInput}
                placeholder={data.placeholder}
                keyboardType="default"
            />
        </>
    )
}