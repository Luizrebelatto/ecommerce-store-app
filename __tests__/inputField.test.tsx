import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import InputField from '../src/components/InputField';

describe('InputField Component', () => {
  it('should render input field and your props', () => {
    const mockOnChangeText = jest.fn();
    
    render(
      <InputField
        label='Title field'
        value="Title Book"
        onChangeText={mockOnChangeText}
        placeholder="Enter a value"
        keyboardType='default'
      />
    );

    const input = screen.getByPlaceholderText("Enter a value");

    expect(input.props.value).toBe("Title Book");
    expect(input.props.keyboardType).toBe("default");
    expect(input.props.placeholder).toBe("Enter a value");
    expect(input.props.label).toBe("Title field");
    expect(screen.getByDisplayValue('Title Book')).toBeDefined()

    fireEvent.changeText(input, "New title");

    expect(mockOnChangeText).toHaveBeenCalledWith("New title");
  });
});
