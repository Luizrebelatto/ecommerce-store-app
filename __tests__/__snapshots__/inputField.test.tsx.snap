// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Structure renders correctly and matches snapshot (toJSON) 1`] = `
[
  <Text
    style={
      {
        "fontWeight": "600",
        "marginBottom": 4,
        "marginTop": 12,
      }
    }
  >
    Title
  </Text>,
  <TextInput
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Title Book"
    style={
      {
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "fontSize": 16,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    value="Value InputField"
  />,
]
`;
